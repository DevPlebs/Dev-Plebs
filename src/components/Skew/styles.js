import styled from "@emotion/styled"

export const SVG = styled.div`
  svg {
    position: absolute;
    bottom: 0;
    width: 100%;
    height: 10vh;
  }
`
