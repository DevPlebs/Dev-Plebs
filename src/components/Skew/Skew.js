import React from "react"
import PropTypes from "prop-types"

import { SVG } from "./styles"

const Skew = ({ path, viewBox, fill }) => (
  <SVG>
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox={viewBox}
      preserveAspectRatio="none"
    >
      <polygon fill={fill} points={path} />
    </svg>
  </SVG>
)

Skew.propTypes = {
  path: PropTypes.string.isRequired,
  viewBox: PropTypes.string.isRequired,
  fill: PropTypes.string.isRequired,
}

Skew.defaultProps = {
  path: "0,100 100,0 100,100",
  viewBox: "0 0 100 100",
  fill: "white",
}

export default Skew
