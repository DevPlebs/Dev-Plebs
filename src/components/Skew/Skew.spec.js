import React from "react"
import { create } from "react-test-renderer"

import Skew from "./Skew"

describe("Skew", () => {
  const testProps = {
    path: "0,100 100,0 100,100",
    viewBox: "0 0 100 100",
    fill: "white",
  }

  it("renders correctly", () => {
    const tree = create(<Skew {...testProps} />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
