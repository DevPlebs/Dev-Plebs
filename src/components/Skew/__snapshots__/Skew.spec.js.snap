// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skew renders correctly 1`] = `
.emotion-0 svg {
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 10vh;
}

<div
  className="emotion-0"
>
  <svg
    preserveAspectRatio="none"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <polygon
      fill="white"
      points="0,100 100,0 100,100"
    />
  </svg>
</div>
`;
